### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by asmirnov911**
Without triangles for now.

### CODE REVIEW

File: course6/toyapp_alexs/.vscode/tasks.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (maciekgajewski) [inline course6/toyapp_alexs/.vscode/tasks.json]: Don't put this into git. This is your private config that reflects your private, local environment.
Add .vscode to your _private_ gitignore file!

File: course6/toyapp_alexs/displaylib/surface3d.cc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#include "vector2d.h"
+#include "vector3d.h"
+
+#include <cassert>
+
+namespace Display {
+
+Surface3D::Surface3D(Surface& surface) : mSurface(surface)
+{
+	mSurface = surface;
```
Comment (maciekgajewski) [inline course6/toyapp_alexs/displaylib/surface3d.cc]: what's going on in here?

File: course6/toyapp_alexs/displaylib/surface3d.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+namespace Display {
+
+class Surface3D 
+{
+public:
+	Surface3D(Surface& surface);
+
+	void Clear(Color color);
+	// TODO ask where it is better to use references and where copy ???
+	void SetPixel(const Vector3D& vector, Color color);
```
Comment (maciekgajewski) [inline course6/toyapp_alexs/displaylib/surface3d.h]:
```
// TODO answer: pass const ref if copy is expensive. Pass by copy if you actually need the copy (but then yout make the copy inside of a function).
// TODO: so in general: pass by reference if copying reference may be cheaper than copying the object
```

File: course6/toyapp_alexs/displaylib/transformation3d.cc [hunk 4]
```diff
@@ -0,0 +1,75 @@
+#include <cmath>
```
Comment (maciekgajewski) [inline course6/toyapp_alexs/displaylib/transformation3d.cc]: Order you headers from the mosty specific. local ones, to the ost generic ones.

So always put the std headers at the end.

File: course6/toyapp_alexs/displaylib/transformation3d.cc [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+#include "vector3d.h"
+#include "transformation3d.h"
+
+namespace Display {
+
+Move3D::Move3D(Vector3D move) : Transformation3D()
```
Comment (maciekgajewski) [inline course6/toyapp_alexs/displaylib/transformation3d.cc]: "move" is a confusing name. Consider "translation"

File: course6/toyapp_alexs/displaylib/transformation3d.cc [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+Vector3D Rotate3D::Apply(Vector3D vertex) const
+{
+    return ApplyZ(ApplyY(ApplyX(vertex)));
+}
+
+Vector3D Rotate3D::ApplyX(Vector3D vertex) const
+{
+    return {
+        vertex.x,
+        cos(mThetaX) * vertex.y - sin(mThetaX) * vertex.z,
```
Comment (maciekgajewski) [inline course6/toyapp_alexs/displaylib/transformation3d.cc]: Use qualified functions: ```std::cos```, ```std::sin``` etc

File: course6/toyapp_alexs/displaylib/transformation3d.h [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    Vector3D Apply(Vector3D vertex) const;
+private:
+    double mThetaX;
+    double mThetaY;
+    double mThetaZ;
+
+    Vector3D ApplyX(Vector3D vertex) const;
+    Vector3D ApplyY(Vector3D vertex) const;
+    Vector3D ApplyZ(Vector3D vertex) const;
+};
```
Comment (maciekgajewski) [inline course6/toyapp_alexs/displaylib/transformation3d.h]: The entire idea behind transformation matrices is that you don't need to implement every operation separately.
You just construct a matrix representing the transformation, and you multiply vector by the matrix.
You could also combine transformation by multiplying matrices.

**Review by maciekgajewski — CHANGES_REQUESTED** [review]
Quite good, but the transformations should be implemented in terms of transformation matrix.

Also, please, don't put your private config into the repo; I can't merge it like this.

File: course6/toyapp_alexs/displaylib/transformation3d.cc (hunk 4 above)
Comment (asmirnov911) [reply course6/toyapp_alexs/displaylib/transformation3d.cc]: ok, we usually do it vice versa.

File: course6/toyapp_alexs/.vscode/tasks.json (hunk 1 above)
Comment (asmirnov911) [reply course6/toyapp_alexs/.vscode/tasks.json]: will be removed

File: course6/toyapp_alexs/displaylib/surface3d.cc (hunk 2 above)
Comment (asmirnov911) [reply course6/toyapp_alexs/displaylib/surface3d.cc]: removed

File: course6/toyapp_alexs/displaylib/surface3d.h (hunk 3 above)
Comment (asmirnov911) [reply course6/toyapp_alexs/displaylib/surface3d.h]: Thank you for the answer!

File: course6/toyapp_alexs/displaylib/transformation3d.cc (hunk 6 above)
Comment (asmirnov911) [reply course6/toyapp_alexs/displaylib/transformation3d.cc]: what's the difference?

Comment (asmirnov911) [reply course6/toyapp_alexs/displaylib/transformation3d.cc]: done

File: course6/toyapp_alexs/displaylib/transformation3d.h (hunk 7 above)
Comment (asmirnov911) [reply course6/toyapp_alexs/displaylib/transformation3d.h]: will do this a bit later

File: course6/toyapp_alexs/displaylib/transformation3d.cc (hunk 5 above)
Comment (asmirnov911) [reply course6/toyapp_alexs/displaylib/transformation3d.cc]: Well, it moves the object. Translation sounds confusing to me, let's discuss during the meeting.

**Comment by asmirnov911** [comment]
The last commit can be reviewed separately. First, change is self contained. Second, I changed IDE and all the formatting changed in the previous commit.

**Pull request merged**